x-pack/plugins/ml/public/application/explorer/explorer_swimlane.tsx

```diff
@@ -633,7 +635,21 @@ export class ExplorerSwimlane extends React.Component<ExplorerSwimlaneProps> {
       if (selectedCellTimes.length > 0) {
         this.highlightOverall(selectedCellTimes);
       }
-      this.maskIrrelevantSwimlanes(!!maskAll);
+      // If filtered fields includes a wildcard search maskAll only if there are no points matching the pattern
+      const wildCardFields =
+        filteredFields !== undefined
+          ? filteredFields.filter(field => /\@kuery-wildcard\@$/.test(field))
+          : [];
+      const substring =
+        wildCardFields.length > 0 ? wildCardFields[0].replace(/\@kuery-wildcard\@$/, '') : null;
+      const hasMatchingPoints =
+        substring !== null &&
+        swimlaneData.points.some(point => {
+          return point.laneLabel.includes(substring);
+        });
```

Comment:
I'd suggest creating a dedicated function `hasMatchingPoints` which receives `filteredFields` and `swimlaneData`  as an input and return a boolean. It'll easy to cover this logic with a unit test.
I also see you filter the entire collection of `filteredFields` but eventually use only the first element to get a substring. Maybe worth to use `find` instead of `filter`?

Reply:
TBH I think it should not be part of this component. It makes more sense to resolve it on a parent level and just provide `maskAll` prop.

Reply:
Good call on using `find` 👌 I'll update that.

I'd prefer to keep the logic in this component as it is behavior specific to this component and makes use of props that are passed in already.

Reply:
Updated in d7d89c050a861c251b660bc520915da7fe057ee8
cc @darnautov

Reply:
thanks @alvarezmelissa87! would be great to have 2 unit tests for `hasMatchingPoints` (for true and false)

Reply:
Added in d92bf5cf49d228672239d8af872688e3aba9757b 🙌 😄

Reply:
thanks a lot for adding tests @alvarezmelissa87! 
nit: test description generally starts with `should`, e.g. something like `test('should return true when...')`, but it can be changed later 🙂